feg/gateway/sbi/sbi_test.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2021 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sbi_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/assert"
+
+	n7_client "magma/feg/gateway/sbi/specs/TS29512NpcfSMPolicyControl"
+	n7 "magma/feg/gateway/sbi/specs/TS29512NpcfSMPolicyControlServer"
+	sbi "magma/feg/gateway/sbi/specs/TS29571CommonData"
+)
+
+type MockPcf struct {
+	ListenAddr string
+	Policies   map[string]n7.SmPolicyControl
+}
+
+func NewMockPcf() *MockPcf {
+	return &MockPcf{
+		Policies: make(map[string]n7.SmPolicyControl),
+	}
+}
+
+func (pcf *MockPcf) PostSmPolicies(ctx echo.Context) error {
+	log.Println("PostSmPolicies")
+	var newPolicy n7.SmPolicyContextData
+	err := ctx.Bind(&newPolicy)
+	if err != nil {
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	var policy n7.SmPolicyControl
+	policy.Context = newPolicy
+
+	pcf.Policies["1234"] = policy
+
+	return ctx.NoContent(http.StatusOK)
+}
+
+// GetSmPoliciesSmPolicyId handles GET /sm-policies/{smPolicyId}
+func (pcf *MockPcf) GetSmPoliciesSmPolicyId(ctx echo.Context, smPolicyId string) error {
+	policy, found := pcf.Policies[smPolicyId]
+	if !found {
+		return ctx.NoContent(http.StatusNotFound)
+	}
+	return ctx.JSON(http.StatusOK, policy)
+}
+
+// PostSmPoliciesSmPolicyIdDelete handles POST /sm-policies/{smPolicyId}/delete
+func (pcf *MockPcf) PostSmPoliciesSmPolicyIdDelete(ctx echo.Context, smPolicyId string) error {
+
+	return ctx.NoContent(http.StatusOK)
+}
+
+// PostSmPoliciesSmPolicyIdUpdate handles POST /sm-policies/{smPolicyId}/update
+func (pcf *MockPcf) PostSmPoliciesSmPolicyIdUpdate(ctx echo.Context, smPolicyId string) error {
+
+	return ctx.NoContent(http.StatusOK)
+}
+
+func TestCreateSMPolicy(t *testing.T) {
+	// Start the server and wait till the server is started
+	mockPcf := startServer(t)
+
+	// Create new N7 client object
+	client, err := n7_client.NewClientWithResponses(fmt.Sprintf("http://%s", mockPcf.ListenAddr))
+
+	assert.NoError(t, err, "Client creation failed")
+
+	// Create a new SM Policy
+	ipAddr := sbi.Ipv4Addr("10.1.2.3")
+	imsi := sbi.Supi("12345678901234")
```

Comment:
Define constants at the top of this files to define those values  like`12345678901234`